### PR TITLE
Linux build

### PR DESCRIPTION
**Pull request by Zemurin**

### CODE REVIEW

**Comment by Zemurin** [comment]
The fronter will now build and run, but I have no idea if it'll run the converter at all since I can't test it - without being able to fire up steam games so they init their folders I don't know even where they should be.

Ignore minor inconsistencies with commonFunctions for now, if this works on a proper computer, I'll migrate all true common functions into commonItems.

File: Fronter/Fronter.depend [hunk 1]
```diff
@@ -0,0 +1,3779 @@
+# depslib dependency file v1.0
```
Comment (Idhrendur) [inline Fronter/Fronter.depend:1]: Is the depend file necessary to be checked in? Isn't it created as some part of the build process, like by CMake or something?

File: Fronter/Source/Utils/LinuxFunctions.cpp [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+std::string UTF16ToUTF8(const std::wstring& theString)
+{
+	std::u16string u16str(theString.begin(), theString.end());
+	std::wstring_convert<std::codecvt_utf8_utf16<char16_t>, char16_t> conversion;
+	return conversion.to_bytes(u16str);
+}
+
+std::optional<std::string> getSteamInstallPath(const std::string& steamID)
+{
+	return std::nullopt;
```
Comment (Idhrendur) [inline Fronter/Source/Utils/LinuxFunctions.cpp:16]: As a future task, we can ask people to confirm the install locations mentioned here: https://askubuntu.com/questions/227502/where-are-steam-games-installed

File: Fronter/Fronter.depend (hunk 1 above)
Comment (Zemurin) [reply Fronter/Fronter.depend:1]: Hm, no, looks like codeblocks file, same as the other one. Will drop it.

File: Fronter/Source/Utils/LinuxFunctions.cpp (hunk 2 above)
Comment (Zemurin) [reply Fronter/Source/Utils/LinuxFunctions.cpp:16]: I'll confirm this myself as soon as I get to a linux computer. I think it's safe to assume Ubuntu is the distro of choice that we should target.

**Pull request merged**